pkg/clusteroperator/list.go

```diff
@@ -0,0 +1,123 @@
+package clusteroperator
+
+import (
+	"context"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+// List returns clusterOperators inventory.
+func List(apiClient *clients.Settings) ([]*Builder, error) {
+	glog.V(100).Info("Listing all clusterOperators")
+
+	coList, err := apiClient.ClusterOperators().List(context.Background(), metaV1.ListOptions{})
+
+	if err != nil {
+		glog.V(100).Infof("Failed to list clusterOperators due to %s", err.Error())
+
+		return nil, err
+	}
+
+	var coObjects []*Builder
+
+	for _, clusterOperator := range coList.Items {
+		copiedCo := clusterOperator
+		coBuilder := &Builder{
+			apiClient:  apiClient,
+			Object:     &copiedCo,
+			Definition: &copiedCo,
+		}
+
+		coObjects = append(coObjects, coBuilder)
+	}
+
+	return coObjects, nil
+}
+
+// WaitForAllClusteroperatorsAvailable check that all clusterOperators are in available state.
+func WaitForAllClusteroperatorsAvailable(apiClient *clients.Settings, timeout time.Duration) (bool, error) {
+	glog.V(100).Info("Waiting for all clusterOperators to be in available state")
+
+	coList, err := List(apiClient)
+	if err != nil {
+		glog.V(100).Infof("Failed to list all clusterOperators due to %s", err.Error())
+
+		return false, err
+	}
+
+	// Wait 5 secs in each iteration before condition function () returns true or errors or times out
+	// after availableDuration
```

Comment:
no need for () after function.
what is availableDuration ?

Reply:
yep, left overs